src/miner_val_heartbeat.erl

```diff
@@ -105,7 +107,15 @@ handle_info({blockchain_event, {add_block, Hash, Sync, _Ledger}},
                                     %% key proposals for this heartbeat
                                     %% hashes of the public keys are included in the HB
                                     %% public and private keys are cached locally
-                                    {EmpKeys, EmpKeyHashes} = generate_poc_keys(Ledger),
+                                    %% but only do this is there are gRPC clients which indicates
+                                    %% that this validator is configured to receive poc receipts
+                                    {EmpKeys, EmpKeyHashes} = case ?BYPASS_GRPC_CHECK orelse length(pg:get_members(<<"activity_check_notification">>)) > 0 of
```

Comment:
for completeness you could expose an API in sibyl_bus "get_members/1".  I dont think we need to support erlang versions < 23 anymore but if we do adding this API and support both PG and PG2 would def be appropriate.

Reply:
Thanks. I added the function to sibyl_bus as suggested (https://github.com/helium/sibyl/pull/75) and pushed an update to miner_val_heartbeat to use that function.